src/main/java/bookstore/controller/OrderController.java

```diff
@@ -0,0 +1,74 @@
+package bookstore.controller;
+
+import bookstore.dto.order.CreateOrderRequestDto;
+import bookstore.dto.order.OrderDto;
+import bookstore.dto.order.UpdateOrderRequestDto;
+import bookstore.dto.orderitem.OrderItemDto;
+import bookstore.service.OrderService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Order API", description = "Endpoints for managing orders")
+@RestController
+@RequiredArgsConstructor
+@Validated
+@RequestMapping(value = "/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("hasAnyRole('USER','ADMIN')")
```

Comment:
in our logic only user has an access to create an order
```suggestion
    @PreAuthorize("hasRole('USER')")
```
I guess it will be better, but it's up to you

Reply:
I'll leave it as my admin has user's granted authorities as well

Reply:
so you can leave  only USER (the same for other controllers)